Mathlib/RingTheory/DedekindDomain/Different.lean

```diff
@@ -288,25 +328,107 @@ variable (I)
 @[simp]
 lemma dual_dual :
     dual A K (dual A K I) = I := by
-  rw [dual_eq_mul_inv, dual_eq_mul_inv (I := I), mul_inv, inv_inv, ← mul_assoc, mul_inv_cancel,
+  rw [dual_eq_mul_inv, dual_eq_mul_inv A K (I := I), mul_inv, inv_inv, ← mul_assoc, mul_inv_cancel,
     one_mul]
   rw [dual_ne_zero_iff]
   exact one_ne_zero
 
-lemma dual_involutive :
-    Function.Involutive (dual A K : FractionalIdeal B⁰ L → FractionalIdeal B⁰ L) := dual_dual
-
-lemma dual_injective :
-    Function.Injective (dual A K : FractionalIdeal B⁰ L → FractionalIdeal B⁰ L) :=
-  dual_involutive.injective
-
 variable {I}
 
 @[simp]
 lemma dual_le_dual :
     dual A K I ≤ dual A K J ↔ J ≤ I := by
-  nth_rewrite 2 [← dual_dual (A := A) (K := K) I]
-  rw [le_dual_iff hJ, le_dual_iff (I := J), mul_comm]
-  rwa [dual_ne_zero_iff]
+  nth_rewrite 2 [← dual_dual A K I]
+  rw [le_dual_iff A K hJ, le_dual_iff A K (I := J) (by rwa [dual_ne_zero_iff]), mul_comm]
+
+variable {A K}
+
+lemma dual_involutive :
```

Comment:
Do you think it is worth to use [InvolutiveStar](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Star/Basic.html#InvolutiveStar) here? For example we get injectivity from the general machinery.

Reply:
I don't think this is possible since it requires a choice of `A` and `K`. And the `Function.Involutive.injective` used below is probably the general machinery.